app/js/views/themes-new.jsx

```diff
@@ -0,0 +1,221 @@
+  constructor(props) {
+    super(props);
+    this.state = {
+      services: [],
+
+      service: null,
+      property: null,
+      actionService: null,
+      actionProperty: null
+    };
+
+    this.foxbox = props.foxbox;
+
+    this.updateService = this.updateService.bind(this);
+  }
+
+  componentDidMount() {
+    this.foxbox.getServices()
+      .then(services => {
+        console.log(services);
+        this.updateService(services);
+      })
+      .catch(console.error.bind(console));
+
+    this.foxbox.addEventListener('service-change', this.updateService);
+  }
+
+  componentWillUnmount() {
+    this.foxbox.removeEventListener('service-change', this.updateService);
+  }
+
+  updateService(services = []) {
+    this.setState({ services });
+
+    this.validateSelectedId();
+  }
+
+  /**
+   * Validate the currently selected service against the list of services.
+   */
+  validateSelectedId(service = this.state.service) {
+    if (service === '0') {
+      service = null;
+    } else if (!this.state.services.find(s => s.id === service)) {
+      service = null;
+    }
+
+    if (service === null) {
+      this.setState({
+        service,
+        property: null,
+        actionService: null,
+        actionProperty: null
+      });
+    }
+  }
+
+  handleServiceSelection(evt) {
+    const service = evt.target.value;
+    this.setState({ service });
+
+    this.validateSelectedId(service);
+  }
+
+  handlePropertySelection(evt) {
+    let property = evt.target.value;
+    if (property !== '0') {
+      this.setState({ property });
+    } else {
+      this.setState({
+        property: null,
+        actionService: null,
+        actionProperty: null
+      });
+    }
+  }
+
+  handleActionServiceSelection(evt) {
+    const actionService = evt.target.value;
+    if (actionService !== '0') {
+      this.setState({ actionService });
+    } else {
+      this.setState({
+        actionService: null,
+        actionProperty: null
+      });
+    }
+  }
+
+  handleActionPropertySelection(evt) {
+    let actionProperty = evt.target.value;
+    if (actionProperty !== '0') {
+      this.setState({ actionProperty });
+    } else {
+      this.setState({
+        actionProperty: null
+      });
+    }
+  }
+
+  handleActionButton() {
+    if (this.state.actionProperty === null) {
+      return;
+    }
+  }
+
+  render() {
+    let className = 'new-theme__select';
+    if (this.state.service !== null) {
+      className += ' new-theme__select--selected';
+    }
+    const optionNodes = this.state.services.map(service => (
+      <option key={service.id} value={service.id}>{service.name}</option>
+    ));
+    let serviceSelector = (
+      <select value={this.state.service}
+              onChange={this.handleServiceSelection.bind(this)}
+              className={className}>
+        <option value="0">Select a device</option>
+        {optionNodes}
+      </select>
+    );
+
+    className = 'new-theme__select';
```

Comment:
Not now, but we'll likely need to migrate to https://github.com/JedWatson/classnames someday since we started to manipulate with classes a lot.